docs/source/openfl/components.rst

```diff
@@ -0,0 +1,54 @@
+.. # Copyright (C) 2020-2021 Intel Corporation
+.. # SPDX-License-Identifier: Apache-2.0
+
+.. _openfl_components:
+
+******
+|productName| core components
+******
+
+.. toctree::
+   :maxdepth: 2
+
+   `Spawning`_
+   `Long-living`_
+
+
+.. _openfl_spawning_components:
+
+Spawning components
+##########
+
+Aggregator
+===========
+
+The aggregator is a short-living entity, which means that its lifespan is limited by experiment execution time. 
+It orchestrates collaborators according to the FL plan and performs model updates aggregation.
+The aggregator is spawned by the Director (described below) when a new experiment is submitted.
+
+
+Collaborator
+=============
+
+Collaborator is also a short living entity, it manages training the model on local data: executes assigned tasks, converts DL framework-specific tensor objects to |productName| inner representation, and exchanges model parameters with the aggregator.
+Converting tensors is done by Framework adapter plugins. |productName| ships with Pytorch and TensorFlow 2 framework adapters, this list will be extended in the future. User is free to implement their adapter for the required DL framework enabling |productName| support for experiments using this framework. The adapter plugin interface is simple: there are two required methods to load and extract tensors from a model and optimizer. Model is loaded with relevant weights before every task and at the end of the training task, weights are extracted to be sent to the central node and aggregated.
+Collaborator instance is created by Envoy (described below) when a new experiment is submitted. Every collaborator is a unique service as it is loaded with a local Shard Descriptor to perform tasks included in an FL experiment.
+
+.. _openfl_ll_components:
+
+Long-living components
+#############
+
+Director
+==========
+
+Director is a long-living entity; it is a central node of the federation and may take in several experiments (with the same data interface). When an experiment is reported director starts an aggregator and sends the experiment data to involved envoys; during the experiment, Director oversees the aggregator and updates the user on the status of the experiment.
+Director runs two services: one for frontend users and another one for envoys. It can distribute an experiment reported with the frontend API across the federation and communicate back a trained model snapshot and metrics.
+Director support several concurrent frontend connections (yet experiments are run one by one)
+
+Envoy
+=========
+
+|productName| comes with another long-existing actor called Envoy. It runs on collaborator machines connected to a *Director*. 
+There is one to one mapping between *Envoys* and Dataset shards: every *Envoy* needs exactly one *Shard Descriptor* to run. 
```

Comment:
This makes the ontology between the Envoy and Datasets very clear. It might also be helpful to provide a link to the ShardDescriptor here

Reply:
I believe the place, where `ShardDescriptor` is explained, is `director-based workflow`, I can place a link to that section.
Alternatively, we could create another section under `OpenFL structure` that would describe the interface entities for `collaborator nodes` owners / managers

Reply:
For now I added a link to the `ShardDescriptor` base class